examples/dit/diffusion/respace.py

```diff
@@ -77,7 +77,7 @@ def __init__(self, use_timesteps, **kwargs):
         new_betas = []
         for i, alpha_cumprod in enumerate(base_diffusion.alphas_cumprod):
             if i in self.use_timesteps:
-                new_betas.append(1 - alpha_cumprod / last_alpha_cumprod)
+                new_betas.append(1 - float(alpha_cumprod) / last_alpha_cumprod)
```

Comment:
it should not introduce any difference, right?

Reply:
Because `base_diffusion.alphas_cumprod` is already a `ms.Tensor`. I notice that in `ms.set_context`, if `precision_mode` is set incorrectly, `alpha_cumprod` maybe a `fp16` number, which can lead to the `new_betas` contain zeros ($1-0.9999=0$ ). I manually force it to be a float value.

Reply:
strange bug? I think `precision_mode` will only affect the calculuation method, but in your case it will also affect the tensor store type

Reply:
Even though I printed `alpha_cumprod` which seemed to be a fp32 value, I still got `1-0.9999=0`. Weird.

Reply:
What do you mean by "`precision_mode` set incorrectly"?

Reply:
For example, setting it to `force_fp16` will lead to this problem. I previously thought this only affected the Tensor computing within nn.Cell, but it did affect Tensors in `GaussianDiffusion` class.

In torch codes, `GaussianDiffusion` class contains only np.array (np.arrays' precision is not relevant to torch.auto_mixed_precision), while in ms codes, all arrays in `GaussianDiffusion` class have been converted to ms.Tensor. I hope to force the precision of the Tensors in `GaussianDiffusion` class.

Reply:
> Even though I printed `alpha_cumprod` which seemed to be a fp32 value, I still got `1-0.9999=0`. Weird.

I think it is cause by the `minus` operation, it is done in fp16, though the two values are both fp32